test/Libraries/Microsoft.Extensions.AI.Evaluation.NLP.Tests/GLEUAlgorithmTests.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.Extensions.AI.Evaluation.NLP.Common;
+using Xunit;
+using static Microsoft.Extensions.AI.Evaluation.NLP.Common.GLEUAlgorithm;
+
+namespace Microsoft.Extensions.AI.Evaluation.NLP.Tests;
+
+public class GLEUAlgorithmTests
+{
+    [Fact]
+    public void TestZeroMatches()
+    {
+        string[][] references = ["The candidate has no alignment to any of the references".Split(' '),];
+        string[] hypothesis = "John loves Mary".Split(' ');
+
+        double score = SentenceGLEU(references, hypothesis);
+        Assert.Equal(0.0, score, 4);
+    }
+
+    [Fact]
+    public void TestFullMatches()
+    {
+        string[][] references = ["John loves Mary".Split(' '),];
+        string[] hypothesis = "John loves Mary".Split(' ');
+
+        double score = SentenceGLEU(references, hypothesis);
+        Assert.Equal(1.0, score, 4);
+    }
+
+    [Fact]
+    public void TestSentenceGLEUExampleA()
+    {
+        string[][] references = [
+            "It is a guide to action that ensures that the military will forever heed Party commands".Split(' '),
+            "It is the guiding principle which guarantees the military forces always being under the command of the Party".Split(' '),
+            "It is the practical guide for the army always to heed the directions of the party".Split(' ')
+        ];
+        string[] hypothesis = "It is a guide to action which ensures that the military always obeys the commands of the party".Split(' ');
+
+        double score = SentenceGLEU(references, hypothesis);
+        Assert.Equal(0.2778, score, 4);
+    }
+
+    [Fact]
+    public void TestSentenceGLEUMilitaryExampleA()
+    {
+        string[][] references = [
+            "It is a guide to action that ensures that the military will forever heed Party commands".Split(' '),
+        ];
+        string[] hypothesis = "It is a guide to action which ensures that the military always obeys the commands of the party".Split(' ');
+
+        double score = SentenceGLEU(references, hypothesis);
+        Assert.Equal(0.43939, score, 4);
+    }
+
+    [Fact]
+    public void TestSentenceGLEUMilitaryExampleB()
```

Comment:
The test names in this file are slightly odd :) I don't feel too strongly about it but one way to avoid this could be to use theories.